chia/rpc/wallet_rpc_api.py

```diff
@@ -1714,6 +1724,26 @@ async def get_offer_summary(self, request) -> EndpointResult:
 
     async def check_offer_validity(self, request) -> EndpointResult:
         offer_hex: str = request["offer"]
+
+        ###
```

Comment:
so we're double parsing offers just to provide the correct error message?

why not expand existing error message to `Invalid offer data or pre 1.7 offer format` or similar?

Reply:
What existing error message are you referring to?  I think if we don't explicitly check here, the offer will just get parsed as nothing offered, nothing requested.